packages/components/src/ListView/Items/Items.component.js

```diff
@@ -36,7 +36,7 @@ class Items extends React.PureComponent {
 		this.renderToggleAllOrItem = this.renderToggleAllOrItem.bind(this);
 		this.renderToggleAll = this.renderToggleAll.bind(this);
 
-		this.hasToggleAll = this.props.items.length > 1;
+		this.hasToggleAll = this.props.items.length > 1 && this.props.toggleAllShown;
```

Comment:
Can you test the props before the length ? It is the most determinant condition.